osquery/tables/applications/posix/carbon_black.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/tables.h>
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/replace.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/property_tree/ini_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace tables {
+
+// Path to the Carbon Black sensor id file
+#define kCbSensorIdFile "/var/lib/cb/sensor.id"
+// Path to the Carbon Black sensor settings file
+#define kCbSensorSettingsFile "/var/lib/cb/sensorsettings.ini"
+// Path to Carbon Black direcotry
+#define kCbDir "/var/lib/cb/"
+
+// Get the Carbon Black sensor ID
+void getSensorId(Row& r) {
+  std::string file_contents;
+  if (!forensicReadFile(kCbSensorIdFile, file_contents).ok()) {
+    return;
+  }
+  // check to make sure we have sane data
+  if (file_contents.length() != 16) {
+    return;
+  }
+
+  unsigned int sensor_id;
+  std::string hex_sensor_id = file_contents.substr(11, 16);
+  std::stringstream converter(hex_sensor_id);
+  converter >> std::hex >> sensor_id;
+  r["sensor_id"] = INTEGER(sensor_id);
+}
+
+// Get settings of the Carbon Black sensor
+void getSensorSettings(Row& r) {
+  if (!boost::filesystem::exists(kCbSensorSettingsFile)) {
```

Comment:
I think we have a `pathExists` that does the same thing (but maybe helps in some other ways, I hope?).